### PR TITLE
Add dagger2 and config files

### PR DESCRIPTION
**Pull request by eddhkim**

### CODE REVIEW

File: app/src/main/java/com/elavon/converge/ElavonConvergeProcessorApplication.java [hunk 1]
```diff
@@ -2,25 +2,10 @@
 
 import android.app.Application;
 
-import com.elavon.converge.core.TransactionManager;
-
+@Deprecated
 public class ElavonConvergeProcessorApplication extends Application {
```
Comment (eddhkim) [inline app/src/main/java/com/elavon/converge/ElavonConvergeProcessorApplication.java]: don't need this anymore. should remove?

File: app/src/main/java/com/elavon/converge/processor/ConvergeClient.java [hunk 2]
```diff
@@ -8,57 +8,30 @@
...
-    private static final int CONNECT_TIMEOUT_MS = 10000;
-    private static final int WRITE_TIMEOUT_MS = 5000;
-    private static final int READ_TIMEOUT_MS = 30000;
     private static final MediaType FORM_URL_ENCODED_TYPE = MediaType.parse("application/x-www-form-urlencoded; charset=utf-8");
 
-    private final String host;
-    private final OkHttpClient client;
-    private final XmlMapper xmlMapper;
-
-    public ConvergeClient() throws Exception{
```
Comment (eddhkim) [inline app/src/main/java/com/elavon/converge/processor/ConvergeClient.java]: all moved to the module

File: app/src/main/res/raw/config_test.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "convergeClient": {
```
Comment (eddhkim) [inline app/src/main/res/raw/config_test.json]: all settings moved to here and above file for test vs live

File: app/src/main/java/com/elavon/converge/inject/AppModule.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+@Module
+public class AppModule {
+
+    private final Context context;
+    private final Config config;
+
+    public AppModule(final Context context) {
+        this.context = context;
+        this.config = new ConfigLoader(context.getResources(), Config.Environment.TEST).load();
```
Comment (eddhkim) [inline app/src/main/java/com/elavon/converge/inject/AppModule.java]: environment is harded coded to `TEST` for now. should be picked up from external input.

File: app/src/main/java/com/elavon/converge/config/Config.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
+package com.elavon.converge.config;
+
+public class Config {
```
Comment (eddhkim) [inline app/src/main/java/com/elavon/converge/config/Config.java]: config class coming from `app/src/main/res/raw/config_live.json` and `app/src/main/res/raw/config_test.json`

File: app/src/main/java/com/elavon/converge/ElavonConvergeProcessorApplication.java (hunk 1 above)
Comment (ppalavilli) [reply app/src/main/java/com/elavon/converge/ElavonConvergeProcessorApplication.java]: we might still need it for session management - so let's leave the basic class for now.

File: app/src/main/java/com/elavon/converge/config/Config.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        this.convergeClient = convergeClient;
+    }
+
+    public static class ConvergeClient {
+
+        private String host;
+        private Integer connectTimeoutMs;
+        private Integer readTimeoutMs;
+        private Integer writeTimeoutMs;
+
```
Comment (ppalavilli) [inline app/src/main/java/com/elavon/converge/config/Config.java]: can you also add retries setting here ?

Comment (ppalavilli) [reply app/src/main/java/com/elavon/converge/config/Config.java]: I meant "max number of retries"

**Pull request merged**